.changes/api-os-path-dir-naming.md

```diff
@@ -0,0 +1,8 @@
+---
+"api": patch
+---
+
+* **Breaking Change** Renamed `tempdir` to `tmpdir` (it was a typo).
+* **Breaking Change** Removed all predefined-paths functions in `path` module to be lowercase for example: `appDir` is now `appdir` (better allignment with NodeJS naming convention and os module).
```

Comment:
what do you think about keeping all changes in this PR except this one?

Reply:
I mean `tempDir` to `tmpDir` is logical because of `/tmp` directory on linux.

The other two are re-exports which I think are okay but we can just move `tmpDir` to path module and remove any paths exports from os module.

Reply:
moving tempDir to path is a breaking change  and re-exporting from path module will not follow the allowList so I will just go ahead and close the PR